lib/shared-worker.js

```diff
@@ -0,0 +1,204 @@
+
+                        // Copy all properties to mochaConfig
+                        for (const key of configKeys) {
+                                // except for reporter
+                                if (key === 'reporter') {
+                                        continue;
+                                }
+
+                                // and merge the globals if they exist.
+                                if (key === 'globals') {
+                                        mochaConfig.globals = mochaConfig.globals.concat(config[key]);
+
+                                        continue;
+                                }
+
+                                mochaConfig[key] = config[key];
+                        }
+
+                        context.mocha.setup(mochaConfig);
+                        delete context.window;
+                },
+                runMocha(clientArguments) {
+                        console.info('[karma-mocha-webworker SharedWorker] Run mocha');
+
+                        if (clientArguments) {
+                                let grep = false;
+
+                                if (Array.isArray(clientArguments)) {
+
+                                        clientArguments.reduce((isGrepArg, arg) => {
+                                                if (isGrepArg) {
+                                                        grep = arg;
+
+                                                        return false;
+                                                }
+
+                                                if (arg === '--grep') {
+                                                        return true; // isGrepArg
+                                                }
+
+                                                let match;
+
+                                                if ((match = /^--grep=(.*)$/.exec(arg))) {
+                                                        grep = match[1];
+                                                }
+
+                                                return false;
+                                        }, false);
+                                }
+                                else if (clientArguments.grep) {
+                                        grep = clientArguments.grep;
+                                }
+
+                                if (grep) {
+                                        console.log('[karma-mocha-webworker SharedWorker] mocha grep:', grep);
+                                        context.mocha.grep(grep);
+                                }
+                        }
+
+                        context.mocha.run(() => {
+                                rpc.notify('mochaRunComplete');
+                        });
+                },
+        });
+
+        // a little hack to make mocha function in a worker context
+        context.window = context; // ( deleted in initializeMocha() )
+
+        rpc.notify('ready');
+};
```

Comment:
There's a lot of code duplication between worker.js and shared-worker.js. We should be able to move most of this duplication to a separate file, and then just `require()` that file (browserify will then combine these files in a single bundle, so the `require()` is sort of fake).

I can easily do this myself after merging your PR though. So no need to spend time on this unless you are feeling adventurous :p